app/controllers/shipit/tasks_controller.rb

```diff
@@ -22,8 +22,14 @@ def show
     end
 
     def create
-      @task = stack.trigger_task(params[:definition_id], current_user)
-      redirect_to [stack, @task]
+      @definition = stack.find_task_definition(params[:definition_id])
+
+      if @definition.allow_concurrency? || params[:force] || !@stack.deploying?
```

Comment:
I'm pretty sure `deploying?` will cause misconception problems to someone someday... just saying. Someone will assume it actually means deploying...